cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java

```diff
@@ -184,15 +182,21 @@ public void deleteVirtualOutlet(String officeId, String projectId, String virtua
         });
     }
 
-    private Outlet mapToOutlet(PROJECT_STRUCTURE_OBJ_T projectStructure,
+    private Outlet mapToOutlet(PROJECT_STRUCTURE_OBJ_T outlet,
                                List<LocationGroup> groups) {
-        Location location = LocationUtil.getLocation(projectStructure.getSTRUCTURE_LOCATION());
-        CwmsId ratingGroupId = getRatingGroupId(location.getName(), groups);
-        CwmsId projectId = LocationUtil.getLocationIdentifier(projectStructure.getPROJECT_LOCATION_REF());
+        Location location = LocationUtil.getLocation(outlet.getSTRUCTURE_LOCATION());
+        LocationGroup locGroupReturn = getRatingGroupId(location.getName(), groups);
```

Comment:
getRatingId can return null, that needs to get handled

Reply:
or is it allowed to be null?

Reply:
it shouldn't be null, but there's no guarantee, and if it's null then the group id and spec should be null.  he handles it correctly, but I think we can use an optional here instead.